### PR TITLE
[Github #322] api key auth

### LINKED ISSUE #322: Support API Key authentication
We currently only support OAuth authentication. API Key is simpler and still used by less-mature APIs. 

#345 suggests interesting solutions for this.

[Spec](https://www.notion.so/nangohq/July-2023-API-Keys-8188de4bed064a0589a8c0976ff12172?pvs=4)

### PR DESCRIPTION
**Pull request by khaliqgant**
Resolves #322 
Resolves https://github.com/NangoHQ/nango/issues/780

- [x] Remove white space changes in docs
- [ ] Remove ts-ignore once frontend version is bumped 
- [x] Create corresponding post hog analytics events
- [x] Send type here? https://github.com/NangoHQ/nango/pull/803/files#diff-e24662cdfc666bf649e1374659ec209bbb0859fa39e116aed413962fee01dbecR136
- [x] Better check here https://github.com/NangoHQ/nango/pull/803/files#diff-38d7d17c71829f59680d38d06daa7498d3621de4601898491bf5903570dc4d27R200

Return types
```
// OAuth 1 & OAuth2
nango.getToken(providerConfigKey: string, connectionId: string, forceRefresh?: boolean): Promise<string>

// API Key
nango.getToken(providerConfigKey: string, connectionId: string, forceRefresh?: boolean): Promise<{ type: 'API_KEY', apiKey: string>

// Basic Auth
nango.getToken(providerConfigKey: string, connectionId: string, forceRefresh?: boolean): Promise<{ type: 'BASIC'; username: string; password: string>

// API Basic Auth
nango.getConnection(providerConfigKey: string, connectionId: string, forceRefresh?: boolean, refreshToken?: boolean)
{
  id: 18,
  created_at: '2023-07-20T11:27:31.446Z',
  updated_at: '2023-07-20T11:27:31.446Z',
  provider_config_key: 'amped!',
  connection_id: 'sda',
  credentials: { type: 'BASIC', username: 'asda', password: 'asda' },
  connection_config: {},
  metadata: null,
  credentials_iv: 'some-value',
  credentials_tag: 'some-value',
  field_mappings: {},
  environment_id: 2
}

// API Key AUth
nango.getConnection(providerConfigKey: string, connectionId: string, forceRefresh?: boolean, refreshToken?: boolean)
{
  id: 18,
  created_at: '2023-07-20T11:27:31.446Z',
  updated_at: '2023-07-20T11:27:31.446Z',
  provider_config_key: 'amped!',
  connection_id: 'sda',
  credentials: { type: 'API_KEY', apiKey: 'asda' },
  connection_config: {},
  metadata: null,
  credentials_iv: 'some-value',
  credentials_tag: 'some-value',
  field_mappings: {},
  environment_id: 2
}


```

### CODE REVIEW

File: docs-v2/api-reference/connection/post.mdx [hunk 1]
```diff
@@ -19,7 +19,6 @@ The required fields depend on the type of token you are trying to import:
 {
 	"connection_id": "<connection-id>",          // Always required
 	"provider_config_key": "<integration-id>",   // Always required
-	"type": "OAUTH2" | "OAUTH1" ,                // OAUTH2 or OAUTH1
```
Comment (khaliqgant) [inline docs-v2/api-reference/connection/post.mdx:22]: @bastienbeurier Import connection no longer needs to specify a type as we can infer that from the provided provider

File: docs-v2/api-reference/integration/get.mdx [hunk 2]
```diff
@@ -6,12 +6,10 @@ openapi: 'GET /config/{providerConfigKey}'
 <ResponseExample>
   ```json Example Response
 {
```
Comment (khaliqgant) [inline docs-v2/api-reference/integration/get.mdx:8]: @bastienbeurier no reason to wrap a single element in an array so I updated this.

File: packages/webapp/src/pages/AuthLink.tsx [hunk 3]
```diff
@@ -33,16 +33,34 @@ export default function AuthLink() {
...
+            };
+        }
+
         nango
             .auth(integrationUniqueKey, connectionId, {
                 user_scope: userScopes,
                 params: params ? JSON.parse(params) : {},
-                authorization_params: authorizationParams ? JSON.parse(authorizationParams) : {}
+                authorization_params: authorizationParams ? JSON.parse(authorizationParams) : {},
+                // @ts-ignore
```
Comment (khaliqgant) [inline packages/webapp/src/pages/AuthLink.tsx:62]: Once the frontend package is published this should be removed

File: packages/webapp/src/pages/ConnectionCreate.tsx [hunk 4]
```diff
@@ -119,12 +126,30 @@ export default function IntegrationCreate() {
...
+        }
+
         nango
             .auth(target.integration_unique_key.value, target.connection_id.value, {
                 user_scope: selectedScopes || [],
                 params: connectionConfigParams || {},
                 authorization_params: authorizationParams || {},
-                hmac: hmacDigest || ''
+                hmac: hmacDigest || '',
+                // @ts-ignore
```
Comment (khaliqgant) [inline packages/webapp/src/pages/ConnectionCreate.tsx:150]: Once the frontend package is published this should be removed

File: scripts/validation/providers/schema.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (khaliqgant) [inline scripts/validation/providers/schema.json:1]: @bastienbeurier this is just a start, but we should expand on this as we go along

**Review by bastienbeurier — APPROVED** [review]
It worked for both Amplitude and Shortcut!

**Pull request merged**